gossipd/gossipd.c

```diff
@@ -1669,6 +1714,35 @@ static struct io_plan *peer_msg_in(struct io_conn *conn,
 	return daemon_conn_read_next(conn, peer->dc);
 }
 
+/* What gossip level do we set for this to meet our target? */
+static enum gossip_level peer_gossip_level(const struct daemon *daemon,
+					   bool gossip_queries_feature)
+{
+	struct peer *peer;
+	size_t gossip_levels[ARRAY_SIZE(gossip_level_targets)];
+	enum gossip_level glevel;
+
+	/* Old peers always give us a flood. */
+	if (!gossip_queries_feature)
+		return GOSSIP_HIGH;
+
+	/* Figure out how many we have at each level. */
+	memset(gossip_levels, 0, sizeof(gossip_levels));
+	list_for_each(&daemon->peers, peer, list)
+		gossip_levels[peer->gossip_level]++;
+
+	/* If we're missing gossip, try to fill GOSSIP_HIGH */
+	if (daemon->gossip_missing != NULL)
+		glevel = GOSSIP_HIGH;
+	else
+		glevel = GOSSIP_MEDIUM;
+
+	while (gossip_levels[glevel] > gossip_level_targets[glevel])
```

Comment:
Shouldn't this be `>=`?

Reply:
Yes, good point.  We also don't provide any insight into what gossip level is for each peer, which would be nice (but means a round-trip to gossipd for listpeers...).

Fixed